packages/venia-ui/lib/components/CartPage/cartPageFragments.js

```diff
@@ -0,0 +1,7 @@
+import gql from 'graphql-tag';
+
+export const CartPageFragment = gql`
+    fragment CartPageFragment on Cart {
+        total_quantity
```

Comment:
Should this fragment just compose itself of the other block components ie:

```js
export const CartPageFragment = gql`
    fragment CartPageFragment on Cart {
        total_quantity
        ...PriceSummaryFragment
        ...PriceAdjustmentsFragment
        ...ProductListingFragment
    }
`
```

Reply:
And then in mutations, instead of requesting individual fragments we could just request the `CartPageFragment`?

```js
export const REMOVE_ITEM_MUTATION = gql`
    mutation removeItem($cartId: String!, $itemId: Int!) {
        removeItemFromCart(input: { cart_id: $cartId, cart_item_id: $itemId }) {
            cart {
                id
                ...CartPageFragment
                ...ProductListingFragment
                ...PriceSummaryFragment
            }
        }
    }
    ${CartPageFragment}
    ${ProductListingFragment}
    ${PriceSummaryFragment}
`;
```

vs
```js
export const REMOVE_ITEM_MUTATION = gql`
    mutation removeItem($cartId: String!, $itemId: Int!) {
        removeItemFromCart(input: { cart_id: $cartId, cart_item_id: $itemId }) {
            cart {
                id
                ...CartPageFragment
            }
        }
    }
    ${CartPageFragment}
`;
```



The former approach is more "slim" in that it requests specific data whereas the latter just does a bulk fetch on mutate.

Reply:
I like the idea of an encapsulating page fragment, which future proofs new features. When new data is needed on the page, a new fragment is added to the page fragment, and any existing mutations automatically pick it up. Easier to maintain, but also relies on the assumption that over fetching doesn't cause unnecessary renders.

Reply:
Seems like over fetching would only cause re-renders if the data changes.